lib/extensions/ar_virtual.rb

```diff
@@ -66,43 +66,99 @@ module ClassMethods
     #
 
     def virtual_delegate(*methods)
-      options = methods.pop
-      unless options.kind_of?(Hash) && options[:to]
+      options = methods.extract_options! || {}
```

Comment:
extract_options! is designed to always give you a hash, so the `|| {}` is not needed.